src/site/content/en/blog/more-capable-form-controls/index.md

```diff
@@ -0,0 +1,305 @@
+class MyCounter extends HTMLElement {
+
+  // Identify the element as a form-associated custom element
+  static formAssociated = true;
+
+
+  constructor() {
+    super();
+    // Get access to the internal form control APIs
+    this._internals = this.attachInternals();
+    // internal value for this control
+    this._value = 0;
+  }
+
+  // Form controls usually expose a "value"  property
+  get value() { return this._value; }
+  set value(v) { this._value = v; }
+
+  // The following properties and methods aren't strictly required, 
+  // but native form controls provide them. Providing them helps
+  // ensure consistency with native controls.
+  get form() { return this._internals.form; }
+  get name() { return this.getAttribute('name'); }
+  get type() { return this.localName; }
+  get validity() {return this.internals_.validity; }
+  get validationMessage() {return this.internals_.validationMessage; }
+  get willValidate() {return this.internals_.willValidate; }
+
+  checkValidity() { return this.internals_.checkValidity(); }
+  reportValidity() {return this.internals_.reportValidity(); }
+
+  ... 
+}
+customElements.register('my-counter', MyCounter);
+```
+
+Once registered, you can use this element wherever you'd use a native form control:
+
+```html
+<form> 
+  <label>Number of bunnies: <my-counter></my-counter></label>
+  <button type="submit">Submit</button>
+</form>
+```
+
+### Setting a value {: #setting-a-value }
+
+The `attachInternals` method returns an `ElementInternals` object that provides access to form control APIs. The most basic of these is the `setFormValue` method, which sets the current value of the control. 
+
+The `setFormValue` method can take one of three types of values:
+
+*   A string value.
+*   A [`File`](https://developer.mozilla.org/en-US/docs/Web/API/File) object.
+*   A [`FormData`](https://developer.mozilla.org/en-US/docs/Web/API/FormData) object. You can use a `FormData` object to pass multiple values (for example, a credit card input control might pass a card number, expiration date, and verification code).
+
+To set a simple value:
+
+```js
+this._internals.setFormValue(this._value);
+```
+
+To set multiple values, you can do something like this:
+
+```js
+// Use the control's name as the base name for submitted data
+const n = this.getAttribute('name');
+const entries = new FormData();
+entries.append(n + '-first-name', this.firstName_);
+entries.append(n + '-last-name', this.lastName_);
+this.internals_.setFormValue(entries);
+```
+
+{% Aside %}The `setFormValue` method takes a second, optional `state` parameter, used to store the internal state of the control. For more information, see [Restoring form state](#restoring-form-state).{% endAside %}
+
+### Input validation
+
+Your control can also participate in form validation by calling the `setValidity`
+method on the internals object.
+
+```js
+// Assume this is called whenever the internal value is updated
+onUpdateValue() {
+  // 
```

Comment:
```suggestion

```
should this be removed?

Reply:
Yeah... Was going to add a comment here, but never got it right, alas.